crates/dojo-core/src/base_test.cairo

```diff
@@ -62,3 +89,59 @@ fn test_upgrade_direct() {
     let upgradeable_dispatcher = IUpgradeableDispatcher { contract_address: base_address };
     upgradeable_dispatcher.upgrade(new_class_hash);
 }
+
+use debug::PrintTrait;
+
+#[starknet::interface]
+trait INameOnly<T> {
+    fn name(self: @T) -> felt252;
+}
+
+#[starknet::contract]
+mod invalid_model {
+    #[storage]
+    struct Storage {}
+
+    #[abi(embed_v0)]
+    impl InvalidModelName of super::INameOnly<ContractState> {
+        fn name(self: @ContractState) -> felt252 {
+            // Pre-computed address of a contract deployed through the world.
+            0x34534b116332dd9459bfde65280822d84c130e3f1faeb63af8455f83e733f4f
+        }
+    }
+}
+
+#[starknet::contract]
+mod invalid_model_world {
+    #[storage]
+    struct Storage {}
+
+    #[abi(embed_v0)]
+    impl InvalidModelName of super::INameOnly<ContractState> {
+        fn name(self: @ContractState) -> felt252 {
+            // World address is 0, and not registered as deployed through the world
+            // as it's itself.
+            0
+        }
+    }
+}
+
+#[test]
+#[available_gas(6000000)]
+#[should_panic(expected: ('invalid model name', 'ENTRYPOINT_FAILED',))]
+fn test_deploy_from_world_invalid_model() {
+    let world = deploy_world();
+
+    let base_address = world.deploy_contract(0, base::TEST_CLASS_HASH.try_into().unwrap());
+    base_address.print();
```

Comment:
```suggestion
```

Reply:
Fixed in 28a083fc7e4f823b6fa0d7f1e9f5e2585ff4af95.

Reply:
In the commit 2c91bc74ef1ceff98e010c5c11ac1dab2f6debbc I addded back the print. It allows to set the address in the `invalid_contract` name function and as the `register_model` takes the gas into account for the salt, removing this line will also cause the address to be different.